source/adapters/cuda/memory.cpp

```diff
@@ -49,7 +50,12 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemBufferCreate(
           cuMemHostRegister(HostPtr, size, CU_MEMHOSTREGISTER_DEVICEMAP));
       AllocMode = BufferMem::AllocMode::UseHostPtr;
     } else if (flags & UR_MEM_FLAG_ALLOC_HOST_POINTER) {
-      UR_CHECK_ERROR(cuMemAllocHost(&HostPtr, size));
+      // UR_CHECK_ERROR(cuMemAllocHost(&HostPtr, size));
+      umf_memory_provider_handle_t umfCUDAprovider =
```

Comment:
name it just "provider" or even call umfMemoryProviderAlloc(hContext->memoryProviderHost, size, 0, &HostPtr);

Reply:
Done